src/main/java/technology/tabula/detectors/RegexSearch.java

```diff
@@ -0,0 +1,641 @@
+	private ArrayList<MatchingArea> _matchingAreas;
+	
+	private Boolean _includeRegexBeforeTable;
+	private Boolean _includeRegexAfterTable;
+
+
+
+	/*
+	 * This constructor is designed to be used for parameters originating in JSON and where no header areas are defined
+	 * NOTE: This constructor will soon be deprecated!!
+	 * @param regexBeforeTable The text pattern that occurs in the document directly before the table that is to be extracted
+	 * @param includeRegexBeforeTable Flag used to include the text pattern in the before the tabular data in the extraction zone
+	 * @param regexAfterTable The text pattern that occurs in the document directly after the table that is to be extracted
+	 * @param includeRegexAfterTable Flag used to include the text pattern directly after the tabular data in the extraction zone
+	 * @param PDDocument The PDFBox model of the PDF document uploaded by the user.
+	 * @param FilteredArea The dimensions of the header and footer margins for the document-can be initialized to null
+	 */
+
+	public RegexSearch(String regexBeforeTable, String includeRegexBeforeTable, String regexAfterTable,
+					   String includeRegexAfterTable, PDDocument document, FilteredArea areaToFilter) {
+
+
+
+		this(regexBeforeTable,Boolean.valueOf(includeRegexBeforeTable),regexAfterTable,
+			Boolean.valueOf(includeRegexAfterTable),document,areaToFilter);
+
+		// System.out.println(includeRegexBeforeTable);
+		// System.out.println(includeRegexAfterTable);
+		// System.out.println(Boolean.valueOf(includeRegexBeforeTable));
+		// System.out.println(Boolean.valueOf(includeRegexAfterTable));
+	}
+
+	public RegexSearch(String regexBeforeTable,Boolean includeRegexBeforeTable, String regexAfterTable,
+					   Boolean includeRegexAfterTable, PDDocument document, FilteredArea filterArea) {
+		_regexBeforeTable = Pattern.compile(regexBeforeTable);
+		_regexAfterTable = Pattern.compile(regexAfterTable);
+
+		_includeRegexBeforeTable = includeRegexBeforeTable;
+		_includeRegexAfterTable = includeRegexAfterTable;
+
+		_matchingAreas = detectMatchingAreas(document, filterArea);
+
+
+	}
+
+	/* getRegexBeforeTable: basic getter function
+	 * @return The regex pattern used to delimit the beginning of the table
+	 */
+	public String getRegexBeforeTable(){
+		return _regexBeforeTable.toString();
+	}
+	/* getRegexAfterTable: basic getter function
+	 * @return The regex pattern used to delimit the end of the table
+	 */
+	public String getRegexAfterTable(){
+		return _regexAfterTable.toString();
+	}
+
+
+	private class SubSectionOfMatch {
+		private Integer _pageNum; //Number of the page that the SubSectionOfMatch is drawn on
+		private Rectangle _area; //Rectangular coordinates defining the boundaries of SubSectionOfMatch
+
+		public SubSectionOfMatch(Integer pageNum, Rectangle area){
+			_pageNum = pageNum;
+			_area = area;
+		}
+
+		public Integer getPageNum(){return _pageNum;}
```

Comment:
Never used. Remove. If used, rename to `getPageNumber` for consistency with other methods.